### PR TITLE
Grease the reserved bits

### LINKED ISSUE #1511: Short header: reserved bits
The test says:

> Reserved: The sixth, seventh, and eighth bits (0x7) of octet 0 are reserved for experimentation.

But there's no guidance on what they should be set to, so I think it should be:

> Reserved: The sixth, seventh, and eighth bits (0x7) of octet 0 are reserved for experimentation and MUST be set to zero.

### PR DESCRIPTION
**Pull request by martinthomson**
Closes #1511.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -444,7 +444,10 @@ Google QUIC Demultipexing Bit:
 Reserved:
 
 : The sixth, seventh, and eighth bits (0x7) of octet 0 are reserved for
-  experimentation.
+  experimentation.  Endpoints can MUST ignore these bits unless they are
```
Comment (marten-seemann) [inline draft-ietf-quic-transport.md]: c̶a̶n̶

gQUIC 44 is using the Long Header according to draft-11 (?), where the last two bits are used to encode the packet number length. So technically, the bits are not ignored there.

Comment (martinthomson) [reply draft-ietf-quic-transport.md]: This is version-specific text, so I think that we're OK.  Even if qQUIC doesn't use the invariants (yet) to identify versions.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -444,7 +444,10 @@ Google QUIC Demultipexing Bit:
 Reserved:
 
 : The sixth, seventh, and eighth bits (0x7) of octet 0 are reserved for
-  experimentation.
+  experimentation.  Endpoints MUST ignore these bits unless they are
```
Comment (larseggert) [inline draft-ietf-quic-transport.md]: s/ignore these bits/ignore these bits on receipt/

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -444,7 +444,10 @@ Google QUIC Demultipexing Bit:
...
+  participating in an experiment that uses these bits.  An endpoint not
+  actively using these bits SHOULD set the value randomly to protect against
```
Comment (larseggert) [inline draft-ietf-quic-transport.md]: s/should set the value randomly/should randomize these bits on transmit/

**Pull request merged**